MessageHandler.js

```diff
@@ -25,14 +27,24 @@ class MessageHandler{
             return;
         }
 
-        // we only accept messages that start with "$"
-        if(!message || message.length < 1 || message[0] !== "$"){
+        // If this message starts with the comment prefix (and one is set)
+        // ignore this message no matter what
+        if (this.commentPrefix.length > 0 && message.startsWith(this.commentPrefix)){
```

Comment:
I'm not very concerned with zero-length strings, but we probably should handle that in some way. I would just:
```
if (this.commentPrefix && this.commentPrefix !== "" && message.startsWith(this.commentPrefix))
```